runtime/lua/vim/treesitter/languagetree.lua

```diff
@@ -409,6 +428,46 @@ function LanguageTree:_add_injections()
   return query_time
 end
 
+--- Run an asynchronous parse, calling {on_parse} when complete.
+---
+--- @param range boolean|Range|nil
+--- @param on_parse fun(trees: TSTrees)
+function LanguageTree:_async_parse(range, on_parse)
+  coroutine.resume(
+    --- @param co_range boolean|Range|nil
+    --- @param co_cb fun(trees: TSTrees)
+    coroutine.create(function(co_range, co_cb)
```

Comment:
Would coroutine.wrap be applicable here?

Reply:
Maybe, but reading the description it looks like using wrap means calling `parse()` again will resume the coroutine, whereas right now it can only be resumed from within itself after a parsing segment is completed. I modeled the `resume(create(...` after coop.nvim's "fire and forget" concept

Also why change? I would guess that at best using `coroutine.wrap` is just the same as now

Reply:
I suggested as a potential simplification since you are starting the coroutine directly after creating it. However I think it's probably best to do this without coroutines, I'll post a patch later.

Reply:
See https://github.com/neovim/neovim/pull/31631#discussion_r1894955052